### PR TITLE
Cache creation & integration.

### PR DESCRIPTION
**Pull request by filipay**
Added some inital tests.
Builds now in Travis CI.
Refactoring and focus on Dependancy Injection for testing.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
   "name": "monitroom",
-  "version": "0.0.0",
+  "version": "0.0.1",
   "private": true,
   "scripts": {
-    "start": "node ./bin/www"
+    "start": "node ./bin/www",
+    "test": "./node_modules/mocha/bin/mocha"
   },
+  "license": "MIT",
```
Comment (HughMcDonnell) [inline package.json]: Is this norm?

File: public/javascripts/dashboard.js [hunk 2]
```diff
@@ -1,30 +1,42 @@
...
+app.get = {};
+app.put = {};
+
+app.get.devices = {};
+
+app.get.devices.api = function(suffix) {
+  var td = $('.td-template').empty().clone();
+  var tr = $('.tr-template').empty().clone();
+  var index = 0;
+  $('.tbody-template').html('<h3>Loading...<\h3>');
```
Comment (HughMcDonnell) [inline public/javascripts/dashboard.js]: Should you setup a CONSTANTS class for strings like this?

File: public/javascripts/table.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
+var DeviceRow = React.createClass({
+  lastSeen : function(timestamp) {
+    var second = 1000;
```
Comment (HughMcDonnell) [inline public/javascripts/table.jsx]: These could also be moved to proposed CONSTANTS class.

File: public/javascripts/table.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  render: function () {
+    if (this.state.isLoading) return <Loading />
+    var rows = [];
+    var index = 0;
+    this.state.data.forEach(function(device) {
+      rows.push(<DeviceRow key={device.mac_addr} device={device} num={index}/>);
+      index++;
+    });
+    return (
+      <table className="table table-striped">
```
Comment (HughMcDonnell) [inline public/javascripts/table.jsx]: Could this stuff be moved some where else to help make it easier to maintain? (separate code from strings)

**Review by HughMcDonnell — APPROVED** [review]
Looks good! Lots of welcome changes. 

General (small) comment in relation to style and formatting: maybe setup some style checking/linting because your style is inconsistent slowing comprehension.

e.g. conditional logic across various files:
if ( cond) 
if (cond) 
if ( cond )

File: public/javascripts/dashboard.js (hunk 2 above)
Comment (filipay) [inline public/javascripts/dashboard.js]: This is more or less deprecated with my latest commit

File: public/javascripts/table.jsx (hunk 3 above)
Comment (filipay) [inline public/javascripts/table.jsx]: I dunno if these should be seen as constants... They're more like singleton's I think. Also a file called constants would get pretty messy and have all sorts of constants. IMO having these in a table class is fine but more input is welcome

File: package.json (hunk 1 above)
Comment (filipay) [inline package.json]: As far as I know yes

File: public/javascripts/table.jsx (hunk 4 above)
Comment (filipay) [inline public/javascripts/table.jsx]: I'm not sure what you mean.. Why should this code be moved?

**Pull request merged**